### PR TITLE
fix(sbb-header): header action animation on hover

### PR DESCRIPTION
**Pull request by federicoisepponfincons**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/master/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes the bug reported by Florian, about missing animation on hover of header action elements
https://github.com/lyne-design-system/lyne-components/issues/1569

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

- added translation to text and icon, when header action is hovered

## Browsers

I tested the build on the following browsers:

- [x] Firefox Desktop
- [x] Chrome Desktop
- [ ] Edge Desktop
- [x] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [ ] VoiceOver Safari Desktop
- [ ] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/components/sbb-header-action/sbb-header-action.scss [hunk 1]
```diff
@@ -46,6 +46,7 @@ $breakpoints: 'zero', 'micro', 'small', 'medium', 'large', 'wide', 'ultra';
 @include sbb.hover-mq($hover: true) {
   :host(:hover) {
     --sbb-header-action-background-color: var(--sbb-color-milk-default);
+    --sbb-header-action-translate: #{sbb.px-to-rem-build(-1)};
```
Comment (jeripeierSBB) [inline src/components/sbb-header-action/sbb-header-action.scss:49]: I know it's not necessary, but I'd suggest to initialize --sbb-header-action-translate-y with 0 on :host{}.

File: src/components/sbb-header-action/sbb-header-action.scss [hunk 2]
```diff
@@ -78,6 +79,15 @@ $breakpoints: 'zero', 'micro', 'small', 'medium', 'large', 'wide', 'ultra';
   }
 }
 
+.sbb-header-action__icon,
+.sbb-header-action__text {
+  color: var(--sbb-color-black-default);
```
Comment (jeripeierSBB) [inline src/components/sbb-header-action/sbb-header-action.scss:84]: setting the color could also work by adding to .sbb-header-action (inherited)

File: src/components/sbb-header-action/sbb-header-action.scss (hunk 1 above)
Comment (jeripeierSBB) [inline src/components/sbb-header-action/sbb-header-action.scss:49]:
```suggestion
    --sbb-header-action-translate-y: #{sbb.px-to-rem-build(-1)};
```

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
I asked Florian for a UX review. It can be that on active, the shift should be reset. But wait for the answer.

**Comment by jeripeierSBB** [comment]
> I asked Florian for a UX review. It can be that on active, the shift should be reset. But wait for the answer.

@federicoisepponfincons after discussion with florian: on active state, the shift should be reset.

**Comment by federicoisepponfincons** [comment]
@jeripeierSBB allright, the shifting should be only on the hover state and not on the active one

**Review by kyubisation — APPROVED** [review]
LGTM 👍

File: src/components/sbb-header-action/sbb-header-action.scss [hunk 3]
```diff
@@ -57,25 +62,41 @@ $breakpoints: 'zero', 'micro', 'small', 'medium', 'large', 'wide', 'ultra';
...
   min-width: var(--sbb-header-action-min-width);
   padding-inline: var(--sbb-header-action-padding-inline);
-  border-radius: var(--sbb-border-radius-16x);
-  transition-property: background-color;
-  transition-duration: var(--sbb-animation-duration-6x);
-  transition-timing-function: ease;
   cursor: pointer;
-  gap: var(--sbb-spacing-fixed-2x);
 
   &:focus-visible {
```
Comment (jeripeierSBB) [inline src/components/sbb-header-action/sbb-header-action.scss:76]: focus-visible state is broken. I think it should also be applied to the ::before (?)

File: src/components/sbb-header-action/sbb-header-action.scss [hunk 4]
```diff
@@ -57,25 +62,41 @@ $breakpoints: 'zero', 'micro', 'small', 'medium', 'large', 'wide', 'ultra';
...
     @include sbb.focus-outline;
   }
+
+  &::before {
+    position: absolute;
+    content: '';
+    inset: var(--sbb-header-action-inset);
+    border-radius: var(--sbb-border-radius-16x);
```
Comment (jeripeierSBB) [inline src/components/sbb-header-action/sbb-header-action.scss:84]: I suggest to use the height as it is a little bit less just a fantasy value.
```suggestion
    border-radius: var(--sbb-header-action-min-height);
```

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
I just realized, that on windows high contrast mode, the header hover doesn't look like other hover effects (e.g. button).

**Comment by federicoisepponfincons** [comment]
@jeripeierSBB added the high contrast effect (as seen in the button component) and fixed the hover / active animation

**Review by jeripeierSBB — APPROVED** [review]
LGTM, thanks!

**Pull request merged**